yarn-project/pxe/src/contract_function_simulator/pxe_oracle_interface.ts

```diff
@@ -296,12 +295,12 @@ export class PXEOracleInterface implements ExecutionDataProvider {
    * so we're keeping it private for now.
    * @param contractAddress - The contract address to silo the secret for
    * @param recipient - The address receiving the notes
-   * @returns A list of indexed tagging secrets
+   * @returns A list of indexed tagging secrets. If the corresponding secret was never used, the index is undefined.
    */
-  async #getIndexedTaggingSecretsForSenders(
+  async #getLastUsedIndexedTaggingSecretsForSenders(
     contractAddress: AztecAddress,
     recipient: AztecAddress,
-  ): Promise<IndexedTaggingSecret[]> {
+  ): Promise<{ secret: DirectionalAppTaggingSecret; index: number | undefined }[]> {
```

Comment:
This is an ugly change but in a followup PR I will basically delete all the tagging sync and do it again so please tolerate.